copy/application.go

```diff
@@ -0,0 +1,98 @@
+package copy
+
+import (
+	"context"
+	"fmt"
+
+	apps "github.com/ninech/apis/apps/v1alpha1"
+	"github.com/ninech/nctl/api"
+	"github.com/ninech/nctl/api/util"
+	"github.com/ninech/nctl/internal/format"
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type applicationCmd struct {
+	resourceCmd
+	Start bool `help:"Automatically start copied app."`
+}
+
+func (app *applicationCmd) Run(ctx context.Context, client *api.Client) error {
+	newApp, err := app.newCopy(ctx, client)
+	if err != nil {
+		return fmt.Errorf("unable to copy app: %w", err)
+	}
+
+	if err := client.Create(ctx, newApp); err != nil {
+		return fmt.Errorf("unable to create Application %q: %w", newApp.GetName(), err)
+	}
+
+	app.printCopyMessage(client, newApp)
+	return nil
+}
+
+func (app *applicationCmd) targetNamespace(client *api.Client) string {
+	if app.TargetProject != "" {
+		return app.TargetProject
+	}
+	return client.Project
+}
+
+func (app *applicationCmd) newCopy(ctx context.Context, client *api.Client) (*apps.Application, error) {
+	oldApp := &apps.Application{}
+	if err := client.Get(ctx, client.Name(app.Name), oldApp); err != nil {
+		return nil, err
+	}
+	newApp := &apps.Application{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      getName(app.TargetName),
+			Namespace: app.targetNamespace(client),
+		},
+		Spec: oldApp.Spec,
+	}
+	newApp.Spec.ForProvider.Paused = !app.Start
+	newApp.Spec.ForProvider.Hosts = []string{}
```

Comment:
Should we also add a flag to allow copying hosts? Or is there any inherent danger that we want to avoid?

Reply:
I think we can have the flag. The DNS verification won't work by default in the new copied app, if the hosts where already verified in the old app. After the corresponding DNS changes have been made in the DNS system the old app still needs to be deleted to make the switch possible.

Reply:
Yeah I figured as much. Added the flag now as I think it will come up at some point anyway.

Reply:
wouldn't copying the secret be more safe? Changing credentials on the new copy will also lead to a change of credentials on the old app, no?